tests/components/mqtt/test_climate.py

```diff
@@ -439,6 +423,44 @@ async def test_set_swing(hass, mqtt_mock):
     assert state.attributes.get("swing_mode") == "on"
 
 
+# CONF_SEND_IF_OFF is deprecated, support will be removed with release 2022.9
+@pytest.mark.parametrize(
+    "send_if_off,assert_async_publish",
+    [
+        (None, [call("swing-mode-topic", "on", 0, False)]),
+        (True, [call("swing-mode-topic", "on", 0, False)]),
+        (False, []),
+    ],
+)
+async def test_set_swing_mode_send_if_off(
+    hass, mqtt_mock, send_if_off, assert_async_publish
+):
+    """Test setting of swing mode if the hvac is off."""
+    config = copy.deepcopy(DEFAULT_CONFIG)
+    if send_if_off is not None:
```

Comment:
Can we change the parameter to be a dict that we can update part of the config with, to avoid the conditional logic?